block/pkg/service/volumes.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2020 The OpenSDS Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package service
+
+import (
+	"context"
+	backendpb "github.com/opensds/multi-cloud/backend/proto"
+	"github.com/opensds/multi-cloud/block/pkg/datastore/driver"
+	"github.com/opensds/multi-cloud/block/pkg/model"
+	pb "github.com/opensds/multi-cloud/block/proto"
+	log "github.com/sirupsen/logrus"
+)
+
+func (b *blockService) getAccessInfo(ctx context.Context, BackendId string) (*backendpb.BackendDetail, error){
+	// get the backedn access_info
+	backendResp, backendErr := b.backendClient.GetBackend(ctx, &backendpb.GetBackendRequest{
+		Id: BackendId})
+
+	if backendErr != nil {
+		log.Infof("Get backend failed with error: %v\n.", backendErr)
+		return nil, backendErr
+	}
+
+	accessInfo := &backendpb.BackendDetail{
+		Region:   backendResp.Backend.Region,
+		Access:   backendResp.Backend.Access,
+		Security: backendResp.Backend.Security,
```

Comment:
Security should not be returned to user

Reply:
It is internal function and not returned to user. Other APIs like create and list will call it for Cloud access. End-user will not get Secret